Spotaut 14860 Added support for aws account create, read ,delete and set credential create, read.

sharadkesarwani opened the pull request:
Added support for AWS account create, read ,delete and set credential create, read.

# Jira Ticket
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-16515

sharadkesarwani committed:
Added support for aws create acct, read acct and delete acct.

Added support for aws create acct, read acct and delete acct.

sharadkesarwani committed:
Added support for aws setCredential

Added support for aws setCredential

sharadkesarwani committed:
Merge branch 'main' into SPOTAUT-14860-support-acctMgmt-apis

sharadkesarwani committed:
removed usage of local sdk

removed usage of local sdk

chandra1-n commented on go.mod:
```diff
@@ -13,4 +13,4 @@ require (
 	github.com/spotinst/spotinst-sdk-go v1.174.0
 	golang.org/x/lint v0.0.0-20200302205851-738671d3881b
-)
+)
```
fix the change in this file otherwise it will not allow to merge

sharadkesarwani replied:
Updated go.mod file with latest SDK version.

sharadkesarwani committed:
Updated latest SDK version and added documentation for account_aws, set_credential_aws

Updated latest SDK version and added documentation for account_aws, set_credential_aws

chandra1-n commented on go.sum:
```diff
@@ -354,6 +354,8 @@ github.com/spf13/pflag v1.0.2/go.mod h1:DYY7MBk1bdzusC3SYhjObp+wFpr4gzcvqqNjLnIn
 github.com/spf13/pflag v1.0.3/go.mod h1:DYY7MBk1bdzusC3SYhjObp+wFpr4gzcvqqNjLnInEg4=
 github.com/spotinst/spotinst-sdk-go v1.174.0 h1:Yuz4oCyrEKFKCDLtiKAAICWZdB2t2s44B59iIRUov+Y=
 github.com/spotinst/spotinst-sdk-go v1.174.0/go.mod h1:Ku9c4p+kRWnQqmXkzGcTMHLcQKgLHrQZISxeKY7mPqE=
```
remove 174 version dependencies

sharadkesarwani replied:
removed

chandra1-n commented on spotinst/config.go:
```diff
@@ -3,6 +3,7 @@ package spotinst
 import (
 	"errors"
 	"fmt"
+	"github.com/spotinst/spotinst-sdk-go/service/account"
```
move this to below, after 27 line

sharadkesarwani replied:
Moved

sharadkesarwani committed:
removed previous SDK version dependency from go.sum

removed previous SDK version dependency from go.sum

chandra1-n commented on spotinst/commons/common_set_credential_aws.go:
```diff
@@ -0,0 +1,88 @@
+)
+
+const (
+	SetCredentialAWSResourceName ResourceName = "spotinst_set_credential_aws"
```
change resource name to spotinst_credentials_aws

sharadkesarwani replied:
updated name to spotinst_credentials_aws

chandra1-n commented on spotinst/commons/common_set_credential_aws.go:
```diff
@@ -0,0 +1,88 @@
+package commons
+
+import (
```
change the file name to common_credentials_aws and use credentials wherever it is credential

sharadkesarwani replied:
updated name to common_credentials_aws.go

chandra1-n commented on docs/resources/set_credential_aws.md:
```diff
@@ -0,0 +1,28 @@
+---
```
change the file name to credentials_aws

sharadkesarwani replied:
updated name to credentials_aws.md

chandra1-n commented on docs/resources/set_credential_aws.md:
```diff
@@ -0,0 +1,28 @@
+---
+layout: "spotinst"
+page_title: "Spotinst: set_credential_aws"
```
change the title credentials_aws

sharadkesarwani replied:
Updated title to credentials_aws

chandra1-n commented on spotinst/commons/common_set_credential_aws.go:
```diff
@@ -0,0 +1,88 @@
+	}
+	return credentialWrapper.GetCredential(), nil
+}
+
```
let's follow same order in all resources: OnCreate, OnRead, NewWrapper, get and set methods

sharadkesarwani replied:
Arranged in order OnCreate, OnRead, NewWrapper, get and set methods.

chandra1-n commented on spotinst/commons/consts.go:
```diff
@@ -187,6 +187,8 @@ const (
 	OceanAKSNPVirtualNodeGroupNodePoolProperties ResourceAffinity = "Ocean_AKS_NP_Virtual_Node_Group_Node_Pool_Properties"
 	OceanAKSNPVirtualNodeGroupVmSizes            ResourceAffinity = "Ocean_AKS_NP_Virtual_Node_Group_Vm_Sizes"
+	AccountAWS                                   ResourceAffinity = "Account_AWS"
+	CredentialAWS                                ResourceAffinity = "Credential_AWS"
```
change to  CredentialsAWS = "Credentials_AWS"

sharadkesarwani replied:
changed to CredentialsAWS = "Credentials_AWS"

sharadkesarwani replied:
changed to CredentialsAWS = "Credentials_AWS"

sharadkesarwani replied:
changed to CredentialsAWS = "Credentials_AWS"

chandra1-n commented on spotinst/credential_aws/consts.go:
```diff
@@ -0,0 +1,8 @@
+package credential_aws
+
```
change folder name to credentials_aws

sharadkesarwani replied:
changes folder name to credentials_aws

chandra1-n commented on spotinst/credential_aws/consts.go:
```diff
@@ -0,0 +1,8 @@
+
+const (
+	IamRole   commons.FieldName = "iamrole"
+	AccountId commons.FieldName = "accountid"
```
use camelcase "accountId"

sharadkesarwani replied:
camelCase is not allowed , modified name to account_id.

chandra1-n commented on spotinst/credential_aws/fields_spotinst_credential_aws.go:
```diff
@@ -0,0 +1,82 @@
+package credential_aws
+
```
change file name

sharadkesarwani replied:
updated file name to fields_spotinst_credentials_aws.go

chandra1-n commented on spotinst/provider.go:
```diff
@@ -89,6 +89,12 @@ func Provider() *schema.Provider {
+			string(commons.AccountAWSResourceName): resourceSpotinstAccountAWS(),
+
+			// AWS set credential
+			string(commons.SetCredentialAWSResourceName): resourceSpotinstCredentialAWS(),
```
CredentialsAWSResourceName and resourceSpotinstCredentialsAWS()

sharadkesarwani replied:
updated to string(commons.CredentialsAWSResourceName): resourceSpotinstCredentialsAWS(),

chandra1-n commented on spotinst/resource_spotinst_account_aws.go:
```diff
@@ -0,0 +1,147 @@
+
+	var resp *aws.CreateAccountOutput = nil
+	input := &aws.CreateAccountInput{Account: account}
+	println(input)
```
remove println

sharadkesarwani replied:
removed

chandra1-n commented on spotinst/resource_spotinst_account_aws.go:
```diff
@@ -0,0 +1,147 @@
+		log.Printf("===> Account create configuration: %s", json)
+	}
+
+	var resp *aws.CreateAccountOutput = nil
```
use output instead of resp

sharadkesarwani replied:
Changed resp to output

chandra1-n commented on spotinst/resource_spotinst_account_aws.go:
```diff
@@ -0,0 +1,147 @@
+		commons.AccountAWSResource.GetName(), id)
+
+	input := &aws.ReadAccountInput{AccountID: spotinst.String(id)}
+	resp, err := meta.(*Client).account.CloudProviderAWS().ReadAccount(context.Background(), input)
```
use output instead of resp

sharadkesarwani replied:
Changed resp to output

chandra1-n commented on spotinst/resource_spotinst_credential_aws.go:
```diff
@@ -0,0 +1,105 @@
+package spotinst
```
change the file name

sharadkesarwani replied:
Updated file name to resource_spotinst_credentials_aws.go

chandra1-n commented on spotinst/resource_spotinst_credential_aws.go:
```diff
@@ -0,0 +1,105 @@
+	"github.com/spotinst/terraform-provider-spotinst/spotinst/commons"
+)
+
+func resourceSpotinstCredentialAWS() *schema.Resource {
```
change it to resourceSpotinstCredentialsAWS

sharadkesarwani replied:
changed to resourceSpotinstCredentialsAWS

chandra1-n commented on spotinst/resource_spotinst_credential_aws.go:
```diff
@@ -0,0 +1,105 @@
+)
+
+func resourceSpotinstCredentialAWS() *schema.Resource {
+	setupCredentialAWSResource()
```
change to setupCredentialsAWSResource

sharadkesarwani replied:
Changed func name to setupCredentialsAWSResource

chandra1-n reviewed (changes requested):
please also fix credentials in sdk as well

chandra1-n commented on docs/resources/account_aws.md:
```diff
@@ -0,0 +1,32 @@
+---
+layout: "spotinst"
+page_title: "Spotinst: account_aws"
+subcategory: "Account"
```
change to Accounts

sharadkesarwani replied:
updated

sharadkesarwani committed:
Changed resource name from spotinst_set_credential_aws to spotinst_credentials_aws and chanaged string credential to credentials

Changed resource name from spotinst_set_credential_aws to spotinst_credentials_aws and changed string credential to credentials at all places

sharadkesarwani committed:
modified attribute name.

modified attribute name.

sharadkesarwani committed:
Merge branch 'main' into SPOTAUT-14860-support-acctMgmt-apis

chandra1-n committed:
Merge branch 'main' into SPOTAUT-14860-support-acctMgmt-apis

sharadkesarwani committed:
Merge branch 'SPOTAUT-14860-support-acctMgmt-apis' of ssh://github.com/spotinst/terraform-provider-spotinst into SPOTAUT-14860-support-acctMgmt-apis

chandra1-n committed:
Revert "Merge branch 'SPOTAUT-14860-support-acctMgmt-apis' of ssh://github.com/spotinst/terraform-provider-spotinst into SPOTAUT-14860-support-acctMgmt-apis"

This reverts commit 6f2d20e2e0b8f3b6a53d67c1c44c4281883d483a, reversing
changes made to d61e3d6dd9c403e05f43ae5ea24c4dde2e990ec2.

chandra1-n committed:
fix go.mod line space

fix go.mod line space

chandra1-n committed:
update CHANGELOG.md

update CHANGELOG.md
